src/components/UploadFile.tsx

```diff
@@ -0,0 +1,160 @@
+import React from 'react';
+import type {StyleProp, ViewStyle} from 'react-native';
+import {View} from 'react-native';
+import type {ValueOf} from 'type-fest';
+import useLocalize from '@hooks/useLocalize';
+import useTheme from '@hooks/useTheme';
+import useThemeStyles from '@hooks/useThemeStyles';
+import * as FileUtils from '@libs/fileDownload/FileUtils';
+import CONST from '@src/CONST';
+import type {FileObject} from './AttachmentModal';
+import AttachmentPicker from './AttachmentPicker';
+import Button from './Button';
+import DotIndicatorMessage from './DotIndicatorMessage';
+import Icon from './Icon';
+import * as Expensicons from './Icon/Expensicons';
+import {PressableWithFeedback} from './Pressable';
+import Text from './Text';
+
+type UploadFileProps = {
+    /** Text displayed on button when no file is uploaded */
+    buttonText: string;
+
+    /** Name of currently uploaded file */
+    uploadedFiles: FileObject[];
+
+    /** Handler that fires when file is selected for upload */
+    onUpload: (files: FileObject[]) => void;
+
+    /** Handler that fires when file is removed */
+    onRemove: (fileUri: string) => void;
+
+    /** Array containing accepted file types */
+    acceptedFileTypes: Array<ValueOf<typeof CONST.API_ATTACHMENT_VALIDATIONS.ALLOWED_RECEIPT_EXTENSIONS>>;
+
+    /** Styles to be assigned to Container */
+    style?: StyleProp<ViewStyle>;
+
+    /** Text to display on error message */
+    errorText?: string;
+
+    /** Function called whenever option changes */
+    onInputChange?: (value: FileObject[]) => void;
+
+    /** Function to set error message */
+    setError: (error: string) => void;
+
+    /** Whether to allow multiple files to be selected. */
+    fileLimit?: number;
+
+    /** The total size limit of the files that can be selected. */
+    totalFilesSizeLimit?: number;
+};
+
+function UploadFile({
+    buttonText,
+    uploadedFiles,
+    onUpload,
+    onRemove,
+    acceptedFileTypes,
+    style,
+    errorText = '',
+    setError,
+    onInputChange = () => {},
+    totalFilesSizeLimit = 0,
+    fileLimit = 0,
+}: UploadFileProps) {
+    const {translate} = useLocalize();
+    const styles = useThemeStyles();
+    const theme = useTheme();
+
+    const handleFileUpload = (files: FileObject[]) => {
+        const totalSize = files.reduce((sum, file) => sum + (file.size ?? 0), 0);
+
+        if (totalFilesSizeLimit) {
+            if (totalSize > totalFilesSizeLimit) {
+                setError(translate('attachmentPicker.sizeExceededWithValue', {maxUploadSizeInMB: totalFilesSizeLimit / (1024 * 1024)}));
+                return;
+            }
+        }
+
+        if (fileLimit) {
+            if (files.length > 0) {
+                if (files.length > fileLimit) {
+                    setError(translate('attachmentPicker.tooManyFiles', {fileLimit}));
+                    return;
```

Comment:
Can we do this instead?
```suggestion
            if (files.length > 0 && files.length > fileLimit) {                    
               setError(translate('attachmentPicker.tooManyFiles', {fileLimit}));
                    return;
```